### DISCUSSION TITLE
Module Menu

### DISCUSSION THREAD

**Discussion by ShamanicArts** [noisedeck/noisedeck-docs#31] [Ideas]
It would be great if users could have more agency in selecting modules and creating their work - on the one hand I love the randomisation of Noisedeck it forces a creative mindset - on the other hand I feel unnecessarily limited by the current implementation - I can visualise a signal chain to create an idea but wether I can actually implement it in a session is completely potluck - sometimes I have a module I want but its at the wrong end of the signal chain or other times I have just spent a bunch of time digging in to the documentation learning about a modules use cases but then do not get the opportunity to experiment with it - I know that presets are one way to access this but ultimately I would love to just be able to drag and drop modules into my deck and move them up and down the chain .

**Comment by jesshewitt** [comment]
Thanks for the feedback! We've discussed this and have decided that allowing a free-form module workflow would introduce too much complexity. We're already running into limitations of what is possible with a cross-platform web app. We do have a desktop version planned, and may revisit the idea for it.

While the current signal chain is somewhat fixed, you can work around this by exporting a gif and re-importing it using the pixy, magic-mashup, or pixy-post modules. We are also open to implementing alternate-role versions of a module (synth -> post, for example) if you have specific requests.

**Reply by VegaDeftwing** [reply]
I think one improvement to the situation may be a "Render & Layer" Button which automatically renders the current view to a gif and places it into pixy internally. This has the less-than-great user experience of combined render times, but I think that's an unavoidable trade off.

**Reply by jesshewitt** [reply]
I was just thinking that something like this would be really useful! I'll have to give some thought as to how to implement it.
